tests/examples/CMakeLists.txt

```diff
@@ -14,20 +14,45 @@ cmake_minimum_required(VERSION 3.0.0)
 # Get and update the LUA_PATH so the scripts can be executed without Awesome.
 execute_process(COMMAND lua -e print\(package.path\) OUTPUT_VARIABLE "LUA_PATH_")
 
+# Make sure the system can be called from the test directory
+if(NOT SOURCE_DIR AND ${CMAKE_CURRENT_SOURCE_DIR} MATCHES "/tests/examples")
+    get_filename_component(SOURCE_DIR
+                           "${CMAKE_CURRENT_SOURCE_DIR}/../.." ABSOLUTE)
+    # Used by .luacov.
+    set(ENV{SOURCE_DIRECTORY} ${SOURCE_DIR})
+else()
+    set(SOURCE_DIR ${CMAKE_SOURCE_DIR})
+endif()
+
+if ($ENV{DO_COVERAGE})
+  execute_process(
+    COMMAND lua -e "require('luacov.runner')(os.getenv('SOURCE_DIRECTORY')..'/.luacov')"
+    RESULT_VARIABLE TEST_RESULT
+    OUTPUT_VARIABLE TEST_OUTPUT
+    ERROR_VARIABLE  TEST_ERROR)
+  if (TEST_ERROR EQUAL 0)
+    set(ENV{DO_COVERAGE} 1)
+  else()
+    message(WARNING "Failed to run luacov.runner, skipping DO_COVERAGE.")
+    message("${TEST_ERROR}")
+    set(ENV{DO_COVERAGE} 0)
+  endif()
+endif()
```

Comment:
....why? I would just let it fail instead instead of semi-silently just not generating coverage data.